README.md

```diff
@@ -6,3 +6,52 @@
 [![Coverage](https://codecov.io/gh/invenia/FeatureDescriptors.jl/branch/main/graph/badge.svg)](https://codecov.io/gh/invenia/FeatureDescriptors.jl)
 [![Code Style: Blue](https://img.shields.io/badge/code%20style-blue-4495d1.svg)](https://github.com/invenia/BlueStyle)
 [![ColPrac: Contributor's Guide on Collaborative Practices for Community Packages](https://img.shields.io/badge/ColPrac-Contributor's%20Guide-blueviolet)](https://github.com/SciML/ColPrac)
+
+FeatureDescriptors.jl is an interface package for describing features used in [models](https://invenia.github.io/Models.jl/stable/), [feature engineering](https://invenia.github.io/FeatureTransforms.jl/stable/), and other data-science workflows.
+
+## Getting Started
+
+A `Descriptor` is an abstract type that provides a way to describe the properties and relationships for a collection of features at a high-level.
+`Descriptor` supertypes define some general properties, such as the required tables or value column, that can be overloaded for specific
+subtypes as required. 
+
+For example, say that some weather data is contained in `weather.csv` file but each column describes a different feature we are interested in, such as `:temperature` or `:humidity`. 
+We can define a general `Weather` `Descriptor` with subtypes `Temperature` and `Humidity` that are loaded from the same table but using the appropriate columns:
+
+```julia
+using FeatureDescriptors
+
+abstract type Weather <: Descriptor end
+
+FeatureDescriptors.tables(::Type{<:Weather}) = ["weather.csv"]
+FeatureDescriptors.value_col(::Type{<:Weather}) = :weather  # dummy
+
+abstract type Temperature <: Weather end
+FeatureDescriptors.value_col(::Type{Temperature}) = :temperature
+
+abstract type Humidity <: Weather end
+FeatureDescriptors.value_col(::Type{Humidity}) = :humidity
+
+```
+
+A more specific instance of a feature can also be defined, such as a `MeanTemperature`, perhaps if that feature requires a different [feature engineering](https://invenia.github.io/FeatureTransforms.jl/stable/) treatment.
+```julia
+abstract type MeanTemperature <: Temperature end
+```
+
+Finally, other useful features might be contained in a different table entirely, but we may stil want to encode its relationship to the others.
+
+```julia
+abstract type Precipitation <: Weather end
+FeatureDescriptors.tables(::Type{<:Precipitation}) = ["rainfall.csv"]
+FeatureDescriptors.value_col(::Type{<:Precipitation}) = :rainfall 
+```
```

Comment:
Overall in this section, I'm wondering if we can say more concretely how the Descriptors could be used downstream. I get to the end and think "ok, how is all that going to be used?"

Reply:
I can include an example (or describe) how a `transform` pipeline could be written that dispatches on the `Descriptor`? This is actually a very relevant use-case for us atm and could tie the two packages together?

Reply:
That sounds good